### ISSUE TITLE
Discarding Draft or Sending Mail sometimes crashes app or sometimes just wont close mail view

### ISSUE THREAD

**Issue by firescript** [EddyVerbruggen/nativescript-email#7]
Hey Eddy, when I discard a draft sometimes the app crashes on my device. This is my code:

```
export function teamEmailMessage(){

    var emails = [];
    model.roster.forEach(function(element){
        if(element.email.length > 0){
            emails.push(element.email);
        }        
    }, this);

    email.compose({
        subject: "Team Message",
        body: "\n\n Sent with TeamReps \n <a href='https://teamreps.co'>https://teamreps.co</a>",
        to: emails,
        appPickerTitle: 'Compose with..' // for Android, default: 'Open with..' 
    }).then(function() {
        console.log("Email composer closed");
    });


}
```

I ran it in debug and didnt even get an error message. It doesnt show anything just exits to the homescreen. _The app is closed and needs to start again_  I'm not entirely sure what would be causing this, as i'm not seeing any error output. The other random thing that happens is it will just stay in the compose view, if you press send it sends the mail, but does not close the compose view. cancel does nothing when in this state.

**Comment by EddyVerbruggen** [comment]
Is this iOS or Android?

**Comment by firescript** [comment]
This is on iOS, its really quite random though. Like right now my device is plugged in and i'm running on livesync and i'm not seeing either of these issues happen.

**Comment by EddyVerbruggen** [comment]
I've seen this happen on the simulator only, not on a device yet. Which iOS version is this (and which device)?

**Comment by firescript** [comment]
I am running 9.3 and its a 6s

**Comment by EddyVerbruggen** [comment]
I have a strong suspicion [this line](https://github.com/EddyVerbruggen/nativescript-email/blob/master/email.ios.js#L25) should be in a broader scope. That `MFMailComposeViewController` may be garbage collected when the composer is open (and your app is in the background). I just remembered a similar issue I've seen in a Cordova plugin.

**Comment by EddyVerbruggen** [comment]
Another cause could be the [animated hiding of the compose](https://github.com/EddyVerbruggen/nativescript-email/blob/master/email.ios.js#L61) and your app perhaps performing a transition at the same time (or showing a dialog).

**Comment by firescript** [comment]
Okay so the second scenario i'm not doing in my app, i'm letting your plugin return to the same view it was initiated from. When i built it for messenger plugin, this is where i allocated:

https://github.com/firescript/nativescript-messenger/blob/master/index.ios.js#L53

**Comment by EddyVerbruggen** [comment]
[Interesting line of code](https://github.com/firescript/nativescript-messenger/blob/master/index.ios.js#L70)
🤔🤔🤔🤔🤔

**Comment by firescript** [comment]
Yeah that addition was from @ivanbuhov pretty sure I was having weird closing issues before. Thats when we added that

**Comment by EddyVerbruggen** [comment]
can you add that locally to this plugin and keep an eye on it perhaps?

**Comment by firescript** [comment]
The initial part needs to be wrapped in a NSObject.extend to create the delegate, it started getting a little over my head when @ivanbuhov took over lol, I can try and play around with it though!

**Comment by EddyVerbruggen** [comment]
Let me know if you can pull it off, otherwise I'll take a look later this week. Thanks!

**Comment by ivanbuhov** [comment]
The idea behind [this line](https://github.com/firescript/nativescript-messenger/blob/master/index.ios.js#L70) is to keep the delegate instance alive. Since the `messageComposeDelegate` is weak it will not retain the delegate instance and after the JS wrapper is collected, the native instance will be deallocated, too. Explicitly retaining and [releasing](https://github.com/firescript/nativescript-messenger/blob/master/index.ios.js#L42) the delegate prevents its preliminary deallocation.

**Comment by grasshoppermouse** [comment]
I'm having similar problems as well. After sending an email, sometimes the app crashes. Other times the app returns to the view it was initiated from, but the screen is frozen. No errors in the console (and this is not run: `console.log("Email composer closed");`). Sometimes it returns from sending an email without errors.

**Comment by EddyVerbruggen** [comment]
@grasshoppermouse Thanks for reminding me of this issue - totally slipped my radar. Just released 1.3.4 which should fix this.
